### PR TITLE
Add kafka input source when writing in foreach batch mode.

### PR DESCRIPTION
**Pull request by Imbruced**
### Problem

Closes: #2867

### Solution

I am not sure if thats not too hacky

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/filters/SparkNodesFilter.java [hunk 1]
```diff
@@ -38,9 +40,17 @@ public SparkNodesFilter(OpenLineageContext context) {
...
-    return getLogicalPlan(context)
-        .map(plan -> plan.getClass().getCanonicalName())
-        .filter(node -> filterNodes.contains(node))
-        .isPresent();
+    return getLogicalPlan(context).filter(this::filterNode).isPresent();
+  }
+
+  private boolean filterNode(LogicalPlan plan) {
+    if (plan.isStreaming()) {
+      return filterNodes.stream()
```
Comment (pawel-big-lebowski) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/filters/SparkNodesFilter.java:48]: would you mind adding some comment here to describe why is this happening?

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java [hunk 2]
```diff
@@ -63,16 +66,18 @@
...
+import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
 import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
 import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
 import org.testcontainers.utility.DockerImageName;
 
 @Slf4j
 @Tag("integration-test")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java:80]: What is the reason for doing this? I am wondering if the same behaviour can be achieved with using `@Testcontainers` annotation as used for example here within `io.openlineage.spark.agent.ColumnLineageIntegrationTest`

**Review by pawel-big-lebowski — COMMENTED** [review]
Some clarifications questions left

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java (hunk 2 above)
Comment (Imbruced) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java:83]: I ve used this only to keep handler object instance per class not per tests to not loose events. Spark still starts session with the same config for open lineage, I can use trick with a file but I think its a little tricky.

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**